core/drivers/crypto/caam/hal/common/hal_ctrl.c

```diff
@@ -128,3 +130,127 @@ void caam_hal_ctrl_inc_priblob(vaddr_t baseaddr)
 	if (val != blob)
 		panic("Written PRIBLOB and read PRIBLOB do not match!");
 }
+
+uint8_t caam_hal_ctrl_get_mpcurve(vaddr_t ctrl_addr)
+{
+	uint32_t val_scfgr = 0;
+
+	/*
+	 * On i.MX8MQ B0, the MP is not usable, hence
+	 * return UINT8_MAX
+	 */
+	if (soc_is_imx8mq_b0_layer())
+		return UINT8_MAX;
+
+	/*
+	 * Verify if the device is closed or not
+	 * If device is closed, check get the MPCurve
+	 */
+	if (snvs_is_device_closed()) {
+		/* Get the SCFGR content */
+		val_scfgr = io_caam_read32(ctrl_addr + SCFGR);
+
+		/* Get the MPCurve field value - 4 bits */
+		val_scfgr = (val_scfgr & BM_SCFGR_MPCURVE) >> BS_SCFGR_MPCURVE;
+
+		/*
+		 * If the device is closed and the MPCurve field is 0
+		 * return UINT8_MAX indicating that there is a problem and the
+		 * MP can not be supported.
+		 */
+		if (!val_scfgr)
+			return UINT8_MAX;
+	}
+
+	return val_scfgr;
+}
+
+TEE_Result caam_hal_ctrl_read_mpmr(vaddr_t ctrl_addr, struct caambuf *mpmr)
+{
+	int i = 0;
+	uint32_t val = 0;
+
+	if (mpmr->length < MPMR_NB_REG) {
+		mpmr->length = MPMR_NB_REG;
+		return TEE_ERROR_SHORT_BUFFER;
+	}
+
+	/* MPMR endianness is reverted between write and read */
+	for (; i < MPMR_NB_REG; i += 4) {
+		val = io_caam_read32(ctrl_addr + MPMR + i);
+		mpmr->data[i] = (uint8_t)((val >> 24) & UINT8_MAX);
+		mpmr->data[i + 1] = (uint8_t)((val >> 16) & UINT8_MAX);
+		mpmr->data[i + 2] = (uint8_t)((val >> 8) & UINT8_MAX);
+		mpmr->data[i + 3] = (uint8_t)(val & UINT8_MAX);
```

Comment:
all `& UINT8_MAX` masking are not needed since explicit casts to `uint8_t`.

Reply:
Ok